runtime/wasm/src/mapping.rs

```diff
@@ -52,14 +53,33 @@ where
                     result_sender,
                 } = request;
 
-                let result = instantiate_module_and_handle_trigger(
-                    valid_module.cheap_clone(),
-                    ctx,
-                    trigger,
-                    host_metrics.cheap_clone(),
-                    timeout,
-                    experimental_features,
-                );
+                let result = panic::catch_unwind(AssertUnwindSafe(|| {
+                    instantiate_module_and_handle_trigger(
+                        valid_module.cheap_clone(),
+                        ctx,
+                        trigger,
+                        host_metrics.cheap_clone(),
+                        timeout,
+                        experimental_features,
+                    )
+                }));
+
+                let result = match result {
+                    Ok(result) => result,
+                    Err(panic_info) => {
+                        let err_msg = if let Some(payload) = panic_info
+                            .downcast_ref::<String>()
+                            .map(String::as_str)
+                            .or(panic_info.downcast_ref::<&str>().copied())
+                        {
+                            anyhow!("Subgraph panicked with message: {}", payload)
+                        } else {
+                            anyhow!("Subgraph panicked with an unknown payload.")
+                        };
+                        println!("Subgraph panicked with message: {}", err_msg);
```

Comment:
Usually we don't want `println!` in the code, we could turn this into a log but it seems just redundant with the error, lets remove this.

Reply:
Got it, will keep that in mind going forward, addressed the changes